Form/Processor/RoleFormProcessor.php

```diff
@@ -0,0 +1,116 @@
+use eZ\Publish\API\Repository\RoleService;
+use EzSystems\PlatformUIBundle\Notification\Notification;
+use EzSystems\PlatformUIBundle\Notification\NotificationPoolInterface;
+use EzSystems\PlatformUIBundle\Notification\TranslatableNotificationMessage;
+use EzSystems\RepositoryForms\Event\FormActionEvent;
+use EzSystems\RepositoryForms\Event\RepositoryFormEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\RouterInterface;
+
+class RoleFormProcessor implements EventSubscriberInterface
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var NotificationPoolInterface
+     */
+    private $notificationPool;
+
+    /**
+     * @var RoleService
+     */
+    private $roleService;
+
+    public function __construct(RouterInterface $router, NotificationPoolInterface $notificationPool, RoleService $roleService)
+    {
+        $this->router = $router;
+        $this->notificationPool = $notificationPool;
+        $this->roleService = $roleService;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            RepositoryFormEvents::ROLE_UPDATE => ['processDefaultAction', -10],
+            RepositoryFormEvents::ROLE_SAVE => ['processSaveRole', -10],
+            RepositoryFormEvents::ROLE_REMOVE_DRAFT => ['processRemoveDraft', -10],
+        ];
+    }
+
+    public function processDefaultAction(FormActionEvent $event)
+    {
+        // TODO: When we have role versioning, save draft here. For now, processSaveRole takes care of saving. Follow-up: EZP-24701
+        //$this->addNotification('role.notification.draft_saved');
+    }
+
+    public function processSaveRole(FormActionEvent $event)
+    {
+        /** @var \EzSystems\RepositoryForms\Data\RoleData $roleData */
+        $roleData = $event->getData();
+        $role = $roleData->role;
+        try {
+            $this->roleService->updateRole($role, $roleData);
+            $event->setResponse(
+                new RedirectResponse($this->router->generate('admin_roleList'))
+            );
+            $this->addNotification('role.notification.published');
+        } catch (UnauthorizedException $e) {
+            $this->addError('role.notification.unauthorized');
+        } catch (InvalidArgumentException $e) {
+            $this->addError('role.notification.invalid_argument');
+        }
+    }
+
+    public function processRemoveDraft(FormActionEvent $event)
+    {
+        // TODO: This will delete the role completely. Must delete only the draft. Follow-up: EZP-24701
+//        $role = $event->getData()->role;
+//        $this->roleService->deleterole($role);
+//        try {
+//            $this->roleService->deleterole($role);
+//        } catch (UnauthorizedException $e) {
+//            $this->addError('role.notification.unauthorized');
+//        }
```

Comment:
Or, maybe rather here. I guess we can detect here if name contains __new__\^s{32} then delete the role so cancel "works".